eclair-core/src/test/scala/fr/acinq/eclair/transactions/TransactionsSpec.scala

```diff
@@ -306,14 +317,68 @@ class TransactionsSpec extends FunSuite with Logging {
+    val remoteHtlcPriv = PrivateKey(hex"a8a8a8a8a8a8a8a8a8a8a8a8a8a8a8a8a8a8a8a8a8a8a8a8a8a8a8a8a8a8a8a8")
+    val commitInput = Funding.makeFundingInputInfo(ByteVector32(hex"a0a0a0a0a0a0a0a0a0a0a0a0a0a0a0a0a0a0a0a0a0a0a0a0a0a0a0a0a0a0a0a0"), 0, Btc(1), localFundingPriv.publicKey, remoteFundingPriv.publicKey)
+
+    // htlc1 and htlc2 are two regular incoming HTLCs with different amounts.
+    // htlc2 and htlc3 have the same amounts and should be sorted according to their scriptPubKey
+    val paymentPreimage1 = ByteVector32(hex"1111111111111111111111111111111111111111111111111111111111111111")
+    val paymentPreimage2 = ByteVector32(hex"2222222222222222222222222222222222222222222222222222222222222222")
+    val paymentPreimage3 = ByteVector32(hex"3333333333333333333333333333333333333333333333333333333333333333")
+    val htlc1 = UpdateAddHtlc(ByteVector32.Zeroes, 0, millibtc2satoshi(MilliBtc(100)).toMilliSatoshi, sha256(paymentPreimage1), CltvExpiry(300), TestConstants.emptyOnionPacket)
+    val htlc2 = UpdateAddHtlc(ByteVector32.Zeroes, 1, millibtc2satoshi(MilliBtc(200)).toMilliSatoshi, sha256(paymentPreimage2), CltvExpiry(300), TestConstants.emptyOnionPacket)
+    val htlc3 = UpdateAddHtlc(ByteVector32.Zeroes, 1, millibtc2satoshi(MilliBtc(200)).toMilliSatoshi, sha256(paymentPreimage3), CltvExpiry(300), TestConstants.emptyOnionPacket)
+    val spec = CommitmentSpec(
+      htlcs = Set(
+        DirectedHtlc(IN, htlc1),
+        DirectedHtlc(IN, htlc2),
+        DirectedHtlc(IN, htlc3)
+      ),
+      feeratePerKw = feeratePerKw,
+      toLocal = millibtc2satoshi(MilliBtc(400)).toMilliSatoshi,
+      toRemote = millibtc2satoshi(MilliBtc(300)).toMilliSatoshi)
+
+    val commitTxNumber = 0x404142434446L
+    val commitTx = {
+      val outputs = makeCommitTxOutputs(true, localDustLimit, localRevocationPriv.publicKey, CltvExpiryDelta(toLocalDelay), localDelayedPaymentPriv.publicKey, remotePaymentPriv.publicKey, localHtlcPriv.publicKey, remoteHtlcPriv.publicKey, spec)
+      val txinfo = makeCommitTx(commitInput, commitTxNumber, localPaymentPriv.publicKey, remotePaymentPriv.publicKey, true, outputs)
+      val localSig = Transactions.sign(txinfo, localPaymentPriv)
+      val remoteSig = Transactions.sign(txinfo, remotePaymentPriv)
+      Transactions.addSigs(txinfo, localFundingPriv.publicKey, remoteFundingPriv.publicKey, localSig, remoteSig)
+    }
+
+    val htlcOut1 :: htlcOut2 :: htlcOut3 :: _ = commitTx.tx.txOut.toList
+
+    assert(htlcOut1.amount == 10000000.sat) // htlc1 first because of the smallest amount (BIP69)
+    assert(htlcOut2.amount == 20000000.sat) // htlc2 and htlc3 have the same amount
+    assert(htlcOut3.amount == 20000000.sat)
+
+    //htlc2 comes first because its pubKeyScript is lexicographically smaller than htlc3's
+    assert(htlcOut2.publicKeyScript.toHex < htlcOut3.publicKeyScript.toHex)
+    assert(htlcOut2.publicKeyScript.toHex == "002001ced9e8dad97b85eb0b7d101f7a79587fa890b79ffa7cf98cff1812444b8fe8")
```

Comment:
Instead of these hardcoded publicKeyScript, why not expose `outputs` from line 363 and use the following asserts:

```scala
assert(htlcOut2.publicKeyScript === outputs.filter(_.specItem == DirectedHtlc(IN, htlc2)).map(_.output.publicKeyScript).head)
assert(htlcOut3.publicKeyScript === outputs.filter(_.specItem == DirectedHtlc(IN, htlc3)).map(_.output.publicKeyScript).head)
```